packages/core/admin-client/src/users/UsersResources.ts

```diff
@@ -14,54 +14,44 @@
  * limitations under the License.
  */
 
-import { Codec } from "mongodb-stitch-core-sdk";
-import { User } from "../Resources";
+import { jsonProperty } from "../JsonMapper";
+import { BasicResource } from "../Resources";
 
 // / Creates a new user for an application
-export interface UserCreator {
-  readonly email: string;
-  readonly password: string;
-}
-
-export class UserCreatorCodec implements Codec<UserCreator> {
-  public decode(from: any): UserCreator {
-    return {
-      email: from.email,
-      password: from.password
-    };
-  }
-
-  public encode(from: UserCreator): object {
-    return {
-      email: from.email,
-      password: from.password
-    };
+export class UserCreator {
+  @jsonProperty("email")
+  public readonly email: string;
+  @jsonProperty("password")
+  public readonly password: string;
+
+  constructor(email: string, password: string) {
+    this.email = email;
+    this.password = password;
   }
 }
 
-// / View of a User of an application
-enum Fields {
-  Id = "_id"
+export class User extends UserCreator {
+  @jsonProperty("_id")
+  public readonly id?: string;
 }
 
-export interface UserResponse {
-  readonly id: string;
+// Resource for a single user of an application
+export class UserResource extends BasicResource<User> {
+  public get: () => Promise<User>;
```

Comment:
[q] are these functions actually defined anywhere? don't these need to call the superclass `_get` and `_remove`?

Reply:
Good catch, will write a test for this.